arelle/ValidateFilingText.py

```diff
@@ -763,15 +770,26 @@ def validateHtmlContent(modelXbrl, referenceElt, htmlEltTree, validatedObjectLab
                         messageCodes=("EFM.6.05.34.externalReference", "EFM.5.02.05.externalReference", "FERC.6.05.34.externalReference", "FERC.5.02.05.externalReference"))
                 if attrTag == "src" and attrValue not in checkedGraphicsFiles:
                     if scheme(attrValue) == "data":
-                        modelXbrl.error(messageCodePrefix + "graphicDataUrl",
-                            _("%(validatedObjectLabel)s references a graphics data URL which isn't accepted '%(attribute)s' for <%(element)s>"),
-                            modelObject=elt, validatedObjectLabel=validatedObjectLabel,
-                            attribute=attrValue[:32], element=eltTag)
-                    elif attrValue.lower()[-4:] not in ('.jpg', '.gif'):
+                        try: # allow embedded newlines
+                            m = imgDataMediaBase64Pattern.match(attrValue)
+                            if (not supportedImageTypes["data-scheme"] or
+                                not m or not m.group(1) or not m.group(2)
+                                or m.group(1)[1:] not in supportedImageTypes["mime-types"]
+                                or m.group(1)[1:] != validateGraphicHeaderType(base64.b64decode(m.group(3)))):
```

Comment:
This would be much easier to understand if you used named regex groups or assigned the groups to local vars with descriptive names